Providers/Modules/Plugins/Antimalware/plugin/collectmcafeeinfo.rb

```diff
@@ -9,34 +9,40 @@
 
 class McAfee
 
-	def self.detect()	
-		begin
-			if !File.file?('/opt/isec/ens/threatprevention/bin/isecav')
-				return false
-			end			
-			detectioncmd = `/opt/isec/ens/threatprevention/bin/isecav --version 2>&1`.lines.map(&:chomp)
+	attr_accessor :detectedPath , :mcafeeName , :mcafeeVersion 
+    def self.findMcAfeePath()
+        paths = ['/opt/McAfee/ens/tp/bin/mfetpcli','/opt/isec/ens/threatprevention/bin/isecav']   
+        for path in paths 
+            if File.file?(path)
+                @detectedPath = path
+                detectioncmd = `#{path} --version 2>&1`.lines.map(&:chomp)
+				@mcafeeName = detectioncmd[0]
+                @mcafeeVersion = detectioncmd[1].split(" : ")[1]
+            end
+        end
+    end 
 
-			if !$?.success? || detectioncmd.nil? || detectioncmd.empty?				
-				return false
-			else
-				mcafeeName = detectioncmd[0]
-				mcafeeVersion = detectioncmd[1].split(" : ")[1]
-				if mcafeeName != "McAfee Endpoint Security for Linux Threat Prevention"					
-					return false
-				elsif mcafeeVersion.split(".")[0].to_i  < 10					
-					return false
-				end
-			end			
-			return true
-		rescue => e					
-			return false			
-		end
-	end
+    def self.detect()  
+        begin
+            mcAfeeExist()
```

Comment:
>mcAfeeExist() [](http://example.com/codeflow?start=12&length=13)

Where is this defined? do you want to use findMcAfeePath() here?

Reply:
good point, I will update this, missed its renaming.

Reply:
Updated this